core/lib/types/src/commitment/mod.rs

```diff
@@ -355,7 +357,7 @@ impl L1BatchAuxiliaryOutput {
 
                 let state_diffs_packed = serialize_commitments(&state_diffs);
                 let state_diffs_hash = H256::from(keccak256(&(state_diffs_packed)));
-                let state_diffs_compressed = compress_state_diffs(state_diffs);
+                let state_diffs_compressed = compress_state_diffs(state_diffs.clone());
```

Comment:
This clone seems unnecessary. Could compress_state_diffs take a reference instead?

Reply:
&mut would work. It's a bit dirty but we really don't want to clone that huge array.